README.md

```diff
@@ -46,162 +46,313 @@ pip install -r requirements.txt
-JWT_ACCESS_TOKEN_EXPIRES_MINUTES=2
-SESSION_TYPE=filesystem
-CLIENT_PUBLIC_KEYS_DIRECTORY=keys/
-SERVER_PRIVATE_KEY=keys/server_private_key.pem
-SERVER_PUBLIC_KEY=keys/server_public_key.pem
-SESSION_LIFETIME_MINUTES=4
-REDIRECT_URL_TO_LOUIS_FRONTEND=http://localhost:3000
-```
+   ```bash
+   openssl rand -hex 32
+   ```
 
-4. Replace `<your_secret_key_here>` with the generated key from step 2.
+3. Populate the following variables in the `.env` file. Example for tests and dev:
+
+   ```env
+   # Server settings
+   MEMBRANE_CORS_ALLOWED_ORIGINS=http://localhost:3000
+   MEMBRANE_LOGGING_LEVEL=DEBUG
+   MEMBRANE_LOGGING_FORMAT=[%(asctime)s] [%(levelname)s] [%(filename)s:%(lineno)d:%(funcName)s] - %(message)s
+   MEMBRANE_HEALTH_MESSAGE=Ok
+
+   # Frontend
+   MEMBRANE_FRONTEND=http://localhost:3000
+
+   # Secrets and Keys
+   MEMBRANE_SECRET_KEY=#your secret key generated at step 2
+   MEMBRANE_CLIENT_PUBLIC_KEYS_DIRECTORY=keys/
+   MEMBRANE_SERVER_PRIVATE_KEY=keys/server_private_key.pem
+   MEMBRANE_SERVER_PUBLIC_KEY=keys/server_public_key.pem
+   MEMBRANE_TEST_APP_PUBLIC_KEY=keys/testapp1_public_key.pem
+   MEMBRANE_TEST_APP_PRIVATE_KEY=keys/testapp1_private_key.pem
+
+   # JWT and Session settings
+   MEMBRANE_JWT_ACCESS_TOKEN_EXPIRE_SECONDS=300
+   MEMBRANE_JWT_EXPIRE_SECONDS=300
+   MEMBRANE_SESSION_LIFETIME_SECONDS=300
+   MEMBRANE_SESSION_COOKIE_SECURE=true
+   MEMBRANE_SESSION_TYPE=null
+   MEMBRANE_TOKEN_BLACKLIST=
+   MEMBRANE_APP_ID_FIELD=app_id
+   MEMBRANE_DATA_FIELD=data
+   MEMBRANE_REDIRECT_URL_FIELD=redirect_url
+   MEMBRANE_EXPIRATION_FIELD=exp
+   MEMBRANE_ENCODE_ALGORITHM=RS256
+   MEMBRANE_TOKEN_TYPE=JWT
+
+   # Email settings
+   MEMBRANE_ALLOWED_EMAIL_DOMAINS_PATTERN=^[a-zA-Z0-9._%+-]+@(?:gc.ca|outlook.com)$
+   MEMBRANE_COMM_CONNECTION_STRING=#your azure communication service connection string
+   MEMBRANE_SENDER_EMAIL=#your azure mailfrom email address
+   MEMBRANE_EMAIL_SUBJECT=Please Verify You Email Address
+   MEMBRANE_EMAIL_SEND_SUCCESS=Valid email address, Email sent with JWT link
+   MEMBRANE_EMAIL_SEND_POLLER_WAIT_TIME=2
+   MEMBRANE_EMAIL_SEND_HTML_TEMPLATE=<html><h1>{}</h1></html>
+
+
+   # Error messages
+   MEMBRANE_GENERIC_500_ERROR_FIELD=error
+   MEMBRANE_GENERIC_500_ERROR=An unexpected error occurred. Please try again later.
+   ```
 
-### Running the App
+### 3. Running the App with Docker
 
 1. Build the Docker image:
 
    ```bash
-   docker build -t flask-app .
+   docker build -t your_app_name .
    ```
 
 2. Set your desired port number:
 
    ```bash
-   export PORT=<your_port_here>
+   export PORT=<your_port_here> WORKERS=<number_of_workers> KEEP_ALIVE=<keep-alive-seconds>
    ```
 
 3. Run the Docker container:
-
    ```bash
-   docker run -v ./keys:/app/keys -p $PORT:$PORT -e PORT=$PORT --env-file .env flask-app
+   docker run -v $(pwd)/keys:/app/keys -p $PORT:$PORT \
+      -e PORT=$PORT \
+      -e WORKERS=$WORKERS \
+      -e KEEP_ALIVE=$KEEP_ALIVE \
+      --env-file .env \
```

Comment:
So why not define the environment variables in .env instead of having them specified on the command-line? duplicate. complicating the use unnecessarily.

Reply:
done. Added to .env as optional